### PR TITLE
Static variables

### PR DESCRIPTION
**Pull request by cchallu**
* Added static exogenous variables to all models.
* Refactor TFT only to include continuous variables.
* Added static_df input to the core.NeuralForecast methods.
* Include usage examples in all models using core.NeuralForecast class.

### CODE REVIEW

File: neuralforecast/models/tft.py [hunk 1]
```diff
@@ -93,28 +89,28 @@ def __init__(self, hidden_size,
 
         self.hidden_size = hidden_size
 
-        self.s_cat_inp_lens  = s_cat_inp_lens
-        self.s_cont_inp_size = s_cont_inp_size
```
Comment (kdgutier) [inline neuralforecast/models/tft.py:97]: Lets take the commented code out, instead of keeping it here.

File: neuralforecast/models/tft.py [hunk 2]
```diff
@@ -127,56 +123,37 @@ def __init__(self, hidden_size,
...
 
         # Static inputs are expected to be equal for all timesteps
         # For memory efficiency there is no assert statement
-        s_cat_inp = s_cat_inp[:,0,:] if s_cat_inp is not None else None
-        s_cont_inp = s_cont_inp[:,0,:] if s_cont_inp is not None else None
+        stat_exog = stat_exog[:,:] if stat_exog is not None else None
 
-        s_inp = self._apply_embedding(s_cat_inp, s_cont_inp,
-                                      cat_emb=self.s_cat_embed,
+        s_inp = self._apply_embedding(stat_exog,
```
Comment (kdgutier) [inline neuralforecast/models/tft.py:150]: Let's protect this calls by naming the inputs.
In future revisions of the code we might change the order of the inputs and we can make this robust.

**Review by kdgutier — CHANGES_REQUESTED** [review]
Really cool!

- Lets add the new `static_df` input to the core methods' docstrings.
- Lets clean TFT's commented outdated code.
- Lets make the calls to the TFT methods, to have the name of the inputs to improve robustness.
- Lets institutionalize the static alignment unit test with nbdev_tests

File: neuralforecast/common/_base_windows.py [hunk 3]
```diff
@@ -173,11 +186,20 @@ def _create_windows(self, batch, step):
...
+            static = batch.get('static', None)
+            static_cols=batch.get('static_cols', None)
+            if static is not None:
+                static = torch.repeat_interleave(static, 
+                                    repeats=windows_per_serie, dim=0)
+
             windows_batch = dict(temporal=windows,
-                                 static=None,
-                                 temporal_cols=temporal_cols)
+                                 temporal_cols=temporal_cols,
```
[the remark this answers is not in the document]
Comment (cchallu) [reply neuralforecast/common/_base_windows.py:200]: We need to implement tests for all temporal, static covariates. We will do a full sprint on these tests in the future.

**Review by kdgutier — APPROVED** [review]

**Pull request merged**